regression/cbmc/union14/main.c

```diff
@@ -0,0 +1,13 @@
+int main()
+{
+  unsigned x;
+  __CPROVER_assume(x > 0);
+  union U {
+    unsigned A[x];
+  };
+  int i, i_before;
+  i_before = i;
+  union U u = *((union U *)&i);
+  i = u.A[0];
+  __CPROVER_assert(i == i_before, "going through union works");
```

Comment:
There is quite a lot going on in this test, esp. in making it endian independent. If you have to modify the PR please can you add a few comments on what is going on.

Reply:
Done.